### PR TITLE
Implement `podLogs` through `StreamInvoke`

### PR DESCRIPTION
**Pull request by hausdorff**

### CODE REVIEW

File: pkg/clients/clients.go [hunk 1]
```diff
@@ -174,6 +177,26 @@ func (dcs *DynamicClientSet) IsNamespacedKind(gvk schema.GroupVersionKind) (bool
 	return false, &NoNamespaceInfoErr{gvk}
 }
 
+type LogClient struct {
+	clientset *kubernetes.Clientset
+}
+
+func NewLogClient(clientConfig *rest.Config) (*LogClient, error) {
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(clientConfig)
```
Comment (lblackstone) [inline pkg/clients/clients.go]: Any reason you can't reuse the existing clientset here? Is this using the typed client instead?

File: pkg/provider/provider.go [hunk 2]
```diff
@@ -566,6 +579,102 @@ func (k *kubeProvider) StreamInvoke(
...
+		//
+
+		logLines := make(chan string)
+		defer close(logLines)
+		done := make(chan error)
+		defer close(done)
+
+		go func() {
+			podLogLines := bufio.NewScanner(podLogs)
+			for podLogLines.Scan() {
```
Comment (lblackstone) [inline pkg/provider/provider.go]: From the `Scan` docs:
> Scan panics if the split function returns too many empty tokens without advancing the input. This is a common error mode for scanners.

Do we need to handle this with a `recover` or something?

File: pkg/clients/clients.go (hunk 1 above)
Comment (hausdorff) [reply pkg/clients/clients.go]: It doesn't support querying, right?

File: pkg/provider/provider.go (hunk 2 above)
Comment (hausdorff) [reply pkg/provider/provider.go]: Should we? Generally in Go one doesn't try to recover from panics, except to relay information about a failure before shutting down gracefully.

**Pull request merged**